### ISSUE TITLE
Language and Classifier Agnostic Usage

### ISSUE THREAD

**Issue by hrzafer** [marcotcr/lime#152]
We use mostly Java with various ML algorithms. Let say I have a vectorized dataset with both labels along with predictions of some classifier `A`. Is there a way to use `lime` just with this kind of tabular data?

**Comment by marcotcr** [comment]
No, we need access to the classifier. When generating explanations for an example, we perturb that example in various ways and use the predictions of the classifier on the perturbed examples.

**Comment by YipingNUS** [comment]
> No, we need access to the classifier. When generating explanations for an example, we perturb that example in various ways and use the predictions of the classifier on the perturbed examples.

Hi, I recently ran into a similar problem. I'm working on text classification. My model is implemented in Java. I wrote a [py4j](https://www.py4j.org/) wrapper and defined a python method that takes the input text, calls the Java service and returns the result as a numpy array. The method works and returns the array representing the probability. However, when I passed the method to LIME, it seems to be accessing a built-in method of the model. (Sorry I don't have the stack trace at hand. I'll be happy to add it later on).
 
So I wonder is it only the example (text in this case) that's perturbed? If so, why do we care about which language is the model implemented as long as we can write a Python wrapper for it?
